### ISSUE TITLE
Failed to index file /filename.JPG: No MIMEType in EXIF data: /filename.JPG

### ISSUE THREAD

**Issue by 4bgt** [pulsejet/memories#948]
**Describe the bug**
Getting an error for some files:
[memories] Fehler: Failed to index file /filename.JPG: No MIMEType in EXIF data: /filename.JPG

**To Reproduce**
uploaded an image with *.JPG or *.png as extension, it fails to index.

**Platform:**

- OS: raspian
- Browser: Firefox
- Memories Version: 6.1.5
- Nextcloud Version: 27.1.4
- PHP Version: 8.2

**Comment by 4bgt** [comment]
![test](https://github.com/pulsejet/memories/assets/35797978/835e358a-3bd1-472c-8047-aea0aa342a05)

I also tried renaming, but sadly it did not 

change anything :(

**Comment by pulsejet** [comment]
Can't reproduce, the file indexes just fine for me.
1. Can you try re-uploading it to your instance
2. Try reindexing with the command line and see if that changes anything `occ memories:index -f --folder <nextcloud-path-to-folder>`
3. Run exiftool manually on this file and see what happens. The admin panel of Memories should show the path of the exiftool binary, `<path-to-binary> <path-to-file>`

**Comment by gramels** [comment]
same issue here NC 28.0.1

  - memories: 6.2.2

samples stil needed?

**Comment by pulsejet** [comment]
same questions as above

**Comment by Vlad-Star** [comment]
I am on 28.0.1 and just wanted to mention, that (thanks to this thread!) the "**No MIMEType in EXIF data**" message in the nextcloud log pointed me to a set of corrupted JPG and MP4 files. None of the Windows tools (VLC, Explorer, Paint) were able to recognize them; exiftool executed from the nextcloud server reported "**File format error**":
```
# exiftool 30122010008.mp4
ExifTool Version Number         : 12.70
File Name                       : 30122010008.mp4
Directory                       : .
File Size                       : 184 MB
File Modification Date/Time     : 2010:12:30 12:28:40-05:00
File Access Date/Time           : 2024:01:22 12:00:35-05:00
File Inode Change Date/Time     : 2024:01:19 09:10:32-05:00
File Permissions                : -rw-r--r--
Error                           : File format error
```
And there is no such problem with the file provided by @4bgt here:
```
[root@www tmp]# exiftool test.jpg
ExifTool Version Number         : 12.70
File Name                       : test.jpg
Directory                       : .
File Size                       : 1582 kB
File Modification Date/Time     : 2023:11:29 19:59:50-05:00
File Access Date/Time           : 2023:11:29 19:59:50-05:00
File Inode Change Date/Time     : 2024:01:22 16:28:41-05:00
File Permissions                : -rw-r--r--
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
Exif Byte Order                 : Little-endian (Intel, II)
Make                            : Panasonic
Camera Model Name               : DMC-FT5
Orientation                     : Horizontal (normal)
X Resolution                    : 180
Y Resolution                    : 180
Resolution Unit                 : inches
Software                        : Ver.1.4
Modify Date                     : 2019:06:12 14:49:04
Artist                          :
Y Cb Cr Positioning             : Co-sited
Exposure Time                   : 1/125
F Number                        : 3.4
Exposure Program                : Program AE
[...]
```

**Comment by Luncheon3462** [comment]
I have the same issue.  on NC28.0.3 and current memories app.   Getting No MIMEType in EXIF data error.   When I download the files in question, the exif data is there (MacOS).

**Comment by pulsejet** [comment]
Nobody has provided a reproducible sample yet

**Comment by Luncheon3462** [comment]
> Nobody has provided a reproducible sample yet

Apologies for being obtuse.   You'd like an example file that has valid exif data but that it appears that memories cannot see the MIMEType?

**Comment by pulsejet** [comment]
Correct. Something that can reproduce the error. It seems pretty likely that these files are either corrupt or have a permissions error (Nextcloud can't read them)

**Comment by Luncheon3462** [comment]
Thank you.   I’m happy to provide whatever I can, just not smart enough to know what to provide or what questions to ask.

If exiftool can see proper exit and the file can be opened and properly read by macOS and fedora, is it still possible to be corrupted?

**Comment by pulsejet** [comment]
> If exiftool can see proper exit and the file can be opened and properly read by macOS and fedora, is it still possible to be corrupted?

Assuming this is the same exiftool as used by Memories (which is placed in the `bin-ext` folder), that would point to a possible permissions issue. Maybe try running exiftool on the failing file as the nextcloud user

**Comment by keesfluitman** [comment]
I get the errors also. Ill check if the files are corrupt. The problem is mostly, that even if i have some corrupted exif data in files. Why would my log have to be flooded with logs about it? I get about 50 per second in my nextcloud syslog.

**Comment by Luncheon3462** [comment]
@pulsejet I changed all the permissions to rwxrwxrwx.  makes no difference.   

There are a few others on the nextcloud forums reporting the same.  

@keesfluitman I also have tens of thousands of error messages in the logs.

**Comment by keesfluitman** [comment]
I've recently deleted accounts of my LDAP backend, and am using OAuth now for them to login, so I had to reindex all files. Maybe that has sth to do with it?

**Comment by Luncheon3462** [comment]
@keesfluitman not for me anyway.   

@pulsejet if you believe this is a file permissions error, what specific permissions are needed to avoid the errors?

**Comment by keesfluitman** [comment]
> @keesfluitman not for me anyway.
> 
> @pulsejet if you believe this is a file permissions error, what specific permissions are needed to avoid the errors?
> 
> @pulsejet if the memories directory is owned by the user 'nextcloud' and group 'www-data' and if photos are located in /home/USERNAME/FOLDER/ will memories throw an error 'failed to index...' in the nextcloud log?

For me, it was indeed the ownership that was wrong. So prolly made the copy as root and it didnt keep the ownership of the original copied file (or during moving).

Now i get this though:
Integrity constraint violation: 1062 Duplicate entry
So have to see how i can solve that now.

**Comment by pulsejet** [comment]
> Integrity constraint violation: 1062 Duplicate entry

Full error?

**Comment by pulsejet** [comment]
Failed indexing will not automatically retry now, which should stop flooding the logs

**Comment by Luncheon3462** [comment]
@pulsejet if the memories directory is owned by the user 'nextcloud' and group 'www-data' and if photos are located in /home/USERNAME/FOLDER/ will memories throw an error 'failed to index...' in the nextcloud log?

**Comment by pulsejet** [comment]
It shouldn't, those seem the right permissions

**Comment by keesfluitman** [comment]
> Integrity constraint violation: 1062 Duplicate entry

```
{"reqId":"5hzvLbK7mtAuqzprTfoS","level":3,"time":"March 10, 2024 11:10:16","remoteAddr":"","user":"--","app":"memories","method":"","url":"--","message":"Failed to index file /6ec0f186-8c7c-4d1e-b6ef-7c561ac1fd85/files/Photos/Videos/20200910_125524.mp4: An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1365932' for key 'memories_fileid_index'","userAgent":"--","version":"28.0.3.2","data":{"app":"memories"}}
```

Background:
I had my FreeIPA server as ldap server setup. But I don't like the fact that accounts are literally deleted, when the LDAP backend is no longer active. So I switched to OAuth with Authentik. Meaning I lost all the accounts, and set them up again by simply moving the files over to the new account. It could be that it didn't clean up the memories indices?
